src/api/app/components/watchlist_component.rb

```diff
@@ -0,0 +1,85 @@
+class WatchlistComponent < ApplicationComponent
+  def initialize(user:, project_name: nil, package_name: nil, bs_request_number: nil)
+    super
+
+    @user = user
+    @project_name = project_name
+    @package_name = package_name
+    @bs_request_number = bs_request_number
```

Comment:
We don't really use those instances in the template, right? I would change it to local variables then.